modules/assim.sequential/R/sda_plotting.R

```diff
@@ -0,0 +1,502 @@
+  blue       <<- col2rgb("blue")
+  alphablue  <<- rgb(blue[1], blue[2], blue[3], 75, max = 255)
+  purple       <<- col2rgb("purple")
+  alphapurple <<- rgb(purple[1], purple[2], purple[3], 75, max = 255)
+  brown       <<- col2rgb("brown")
+  alphabrown <<- rgb(brown[1], brown[2], brown[3], 75, max = 255)
+}
+
+
+#internal functions for plotting SDA outputs. Interactive, bias plots as well as post analysis time series both in base plotting and ggplot
+##' @param settings  pecan standard settings list.  
+##' @param t current time - int number giving the position of the current time in obs.time. 
+##' @param obs.time vector of dates of measurements
+##' @param obs.mean list of vectors of the means of observed data named by the measured date.
+##' @param obs.cov list of cov matrices of the observed data named by the measured date.
+##' @param obs list containing the mean and cov object
+##' @param X  dataframe of state variables for each ensemble
+##' @param FORECAST dataframe of state variables for each ensemble
+##' @param ANALYSIS  vector of mean of state variable after analysis
+##' @param plot.title character giving the title for post visualization ggplots
+##' @export
+interactive.plotting.sda<-function(settings,t,obs.times,obs.mean,obs.cov,obs,X,FORECAST,ANALYSIS){
+  #Defining some colors
+  generate_colors_sda()
+  t1         <- 1
+  var.names <- unlist(sapply(settings$state.data.assimilation$state.variable, 
+                             function(x) {
+                               x$variable.name
+                             }, 
+                             USE.NAMES = FALSE), 
+                      use.names = FALSE)
+  #----
+  t1 <- 1
+  names.y <- unique(unlist(lapply(obs.mean[t1:t], function(x) { names(x) })))
+  Ybar <- t(sapply(obs.mean[t1:t], function(x) {
+    tmp <- rep(NA, length(names.y))
+    names(tmp) <- names.y
+    mch <- match(names(x), names.y)
+    tmp[mch] <- x[mch]
+    tmp
+  }))
+  
+  if(any(obs)){
+    Y.order <- na.omit(pmatch(colnames(X), colnames(Ybar)))
+    Ybar <- Ybar[,Y.order]
+    Ybar[is.na(Ybar)] <- 0
+    YCI <- t(as.matrix(sapply(obs.cov[t1:t], function(x) {
+      if (length(x)<2) {
+        rep(NA, length(names.y))
+      }
+      sqrt(diag(x))
+    })))
+    
+    YCI <- YCI[,Y.order]
+    YCI[is.na(YCI)] <- 0
+    
+  }else{
+    YCI <- matrix(NA,nrow=length(t1:t), ncol=max(length(names.y),1))
+  }
+  
+  par(mfrow = c(2, 1))
+  colmax<-2
+  for (i in 1:ncol(FORECAST[[t]])) { #
+    
+    Xbar <- plyr::laply(FORECAST[t1:t], function(x) { mean(x[, i]/rowSums(x[,1:colmax]), na.rm = TRUE) })
+    Xci  <- plyr::laply(FORECAST[t1:t], function(x) { quantile(x[, i]/rowSums(x[,1:colmax]), c(0.025, 0.975), na.rm = TRUE) })
+    
+    Xa <- plyr::laply(ANALYSIS[t1:t], function(x) { mean(x[, i]/rowSums(x[,1:colmax]), na.rm = TRUE) })
+    XaCI <- plyr::laply(ANALYSIS[t1:t], function(x) { quantile(x[, i]/rowSums(x[,1:colmax]), c(0.025, 0.975), na.rm = TRUE) })
+    
```

Comment:
is there a reason why these are different than the previous version (there was no division bu `rowSums()` before)?

```
        Xbar <- plyr::laply(FORECAST[t1:t], function(x) { mean(x[, i], na.rm = TRUE) })
        Xci  <- plyr::laply(FORECAST[t1:t], function(x) { quantile(x[, i], c(0.025, 0.975), na.rm = TRUE) })
        
        Xa <- plyr::laply(ANALYSIS[t1:t], function(x) { mean(x[, i], na.rm = TRUE) })
        XaCI <- plyr::laply(ANALYSIS[t1:t], function(x) { quantile(x[, i], c(0.025, 0.975), na.rm = TRUE) })
```

Reply:
This is interesting, I have no memory of changing this and why I did that .

Reply:
I'm gonna go back to what it was before.